packages/core/src/blocks/HeadingBlockContent/HeadingBlockContent.ts

```diff
@@ -153,6 +156,39 @@ const HeadingBlockContent = createStronglyTypedTiptapNode({
       this.options.domAttributes?.inlineContent || {},
     );
   },
+
+  addNodeView() {
+    return ({ node, HTMLAttributes, getPos }) => {
+      const renderedElement = createDefaultBlockDOMOutputSpec(
+        this.name,
+        `h${node.attrs.level}`,
+        {
+          ...(this.options.domAttributes?.blockContent || {}),
+          ...HTMLAttributes,
+        },
+        this.options.domAttributes?.inlineContent || {},
+      );
+
+      const contentDOM = renderedElement.contentDOM;
+      const contentDOMParentElement = contentDOM.parentElement!;
+
+      const editor = this.options.editor;
+      const block = getBlockFromPos(getPos, editor, this.editor, this.name);
+
+      const toggleWrapper = createToggleWrapper(block as any, editor, {
+        dom: contentDOM,
+        contentDOM,
+      });
```

Comment:
Might be missing something here, but this feels wrong? Why would you need to pass it in twice? Can this not just be a default if `dom` is needed? I'd expect toggleWrapper to completely wrap the element, as in, it should be the bare return here: `return toggleWrapper`